BE/toDoListAPI/src/main/java/school/mindera/toDoListAPI/service/TaskService.java

```diff
@@ -93,20 +95,33 @@ public ResponseEntity<DTOTaskPreview> addTask(DTONewTask newTask) {
         Optional<UsersEntity> user = usersRepository.findById(newTask.getUserId());
         Optional<TasksEntity> parent = Optional.ofNullable(null);
 
-        if (!isNull(newTask.getParentId())){
+        SimpleDateFormat formatData = new SimpleDateFormat("dd/MM/yyyy");
+        Date date;
+        if(!isNull(newTask.getDate())) {
+            try {
+                date = formatData.parse(newTask.getDate());
+            } catch (Exception e) {
+                throw new InvalidTaskException("Date is not valid");
+            }
+        }else{
+            date = null;
+        }
```

Comment:
esse else é completamente desnecessário
```suggestion
        Date date = null;
        if(!isNull(newTask.getDate())) {
            try {
                date = formatData.parse(newTask.getDate());
            } catch (Exception e) {
                throw new InvalidTaskException("Date is not valid");
            }
        }
```

Reply:
e sinceramente acho que não precisas do parse, tipo mandando a data direto ja tava dando

Reply:
o problema é a data que está sendo retornada pro FE, está no formato do java

Reply:
Mas desde o inicio se propos que o front so ia trabalhar com strings, em nenhum momento se falou para trocar isso

Reply:
Vou fazer essa alteracao, para o front receber string